EIPS/eip-4844.md

```diff
@@ -0,0 +1,564 @@
+
+```python
+class BlobTransactionNetworkWrapper(Container):
+    tx: SignedBlobTransaction
+    # KZGCommitment = Bytes48
+    blob_kzgs: List[KZGCommitment, MAX_TX_WRAP_KZG_COMMITMENTS]
+    # BLSFieldElement = uint256
+    blobs: List[Vector[BLSFieldElement, FIELD_ELEMENTS_PER_BLOB], LIMIT_BLOBS_PER_TX]
+```
+
+We do network-level validation of `BlobTransactionNetworkWrapper` objects as follows:
+
+```python
+def validate_blob_transaction_wrapper(wrapper: BlobTransactionNetworkWrapper):
+    commitments = wrapper.tx.message.blob_commitments
+    kzgs = wrapper.blob_kzgs
+    blobs = wrapper.blobs
+    assert len(commitments) == len(kzgs) == len(blobs)
+    for commitment, kzg, blob in zip(commitments, kzgs, blobs):
+        # note: assert blob is not malformatted
+        assert kzg == blob_to_kzg(blob)
+        assert commitment == kzg_to_commitment(kzg)
+```
+
+## Rationale
+
+### On the path to sharding
+
+This EIP introduces blob transactions in the same format in which they are expected to exist in the final sharding specification.
+This provides a temporary but significant scaling relief for rollups by allowing them to scale to 2 MB per slot,
+with a separate fee market allowing fees to be very low while usage of this system is limited.
+
+The core goal of rollup scaling stopgaps is to provide temporary scaling relief,
+without imposing extra development burdens on rollups to take advantage of this relief.
+Today, rollups use calldata. In the future, rollups will have no choice but to use sharded data (also called "blobs")
+because sharded data will be much cheaper.
+Hence, rollups cannot avoid making a large upgrade to how they process data at least once along the way.
+But what we _can_ do is ensure that rollups need to _only_ upgrade once.
+This immediately implies that there are exactly two possibilities for a stopgap: (i) reducing the gas costs of existing calldata,
+and (ii) bringing forward the format that will be used for sharded data, but not yet actually sharding it.
+EIP-4488 is a solution of category (i); this is a solution of category (ii).
+
+The main tradeoff in designing this EIP is that of implementing more now versus having to implement more later:
+do we implement 25% of the work on the way to full sharding, or 50%, or 75%?
+
+The work that is already done in this EIP includes:
+
+* A new transaction type, of the exact same format that will need to exist in "full sharding"
+* _All_ of the execution-layer logic required for full sharding
+* _All_ of the execution / consensus cross-verification logic required for full sharding
+* Layer separation between `BeaconBlock` verification and data availability sampling blobs
+* Most of the `BeaconBlock` logic required for full sharding
+* A self-adjusting independent gasprice for blobs
+  ([multidimensional EIP 1559](https://ethresear.ch/t/multidimensional-eip-1559/11651) with an
+  [exponential pricing rule](https://ethresear.ch/t/make-eip-1559-more-like-an-amm-curve/9082))
```

Comment:
External links shouldn't be included in an EIP.  The EIP should stand on its own.  I recommend moving these links to the discussions-to URL (perhaps in the original post) and changing this to:
```suggestion
* A self-adjusting independent gasprice for blobs
```